src/aolayer.cpp

```diff
@@ -308,7 +308,7 @@ void AOLayer::start_playback(QString p_image)
     frame = 0;
   last_max_frames = max_frames;
   max_frames = m_reader.imageCount();
-  if (((continuous) && (max_frames != last_max_frames)) || max_frames == 0) {
+  if (((continuous) && (max_frames != last_max_frames)) || max_frames == 0 || frame >= max_frames) {
```

Comment:
What is the point of the conditional in line 307

Reply:
if the image isn't being played continuously, we always want to restart it from the beginning so frame is set to zero. the conditional on line 311 handles some more cases where we want to start the image from frame 0:
- if we want to play this continuously, but we can't because the frame counts are different
- if the frame count is zero, meaning this is a static image
- if our target frame is outside the maximum bounds of the frame vectors

Reply:
went ahead and combined the one on 307 into the one now on 309